mllib/src/main/scala/org/apache/spark/ml/feature/ElementwiseProduct.scala

```diff
@@ -55,6 +55,11 @@ class ElementwiseProduct(override val uid: String)
     elemScaler.transform
   }
 
+  override protected def validateInputType(inputType: DataType): Unit = {
+    super.validateInputType(inputType)
+    require(inputType.isInstanceOf[VectorUDT], s"Input type must be VectorUDT but got $inputType.")
```

Comment:
This same pattern occurs in multiple places now. We have `SchemaUtils.checkColumnType`, so I think we should create a new method `SchemaUtils.checkDataType` in a similar manner, to harmonize the functionality and in particular the default error message across all present and future usages.